core/src/test/java/apoc/graph/GraphsTest.java

```diff
@@ -136,34 +130,37 @@ public void testFromCypher() {
     public void testFromDocument() throws Exception {
         Map<String, Object> artistGenesisMap = Util.map("type", "artist", "name", "Genesis", "id", 1L);
         Map<String, Object> albumGenesisMap = Util.map("type", "album", "producer", "Jonathan King", "id", 1L, "title", "From Genesis to Revelation");
-        Map<String, Object> artistGenesisMapExt = new HashMap() {{
+        Map<String, Object> artistGenesisMapExt = new HashMap<>() {{
             putAll(artistGenesisMap);
-            put("albums", Arrays.asList(albumGenesisMap));
+            put("albums", List.of(albumGenesisMap));
         }};
 
         TestUtil.testResult(db, "CALL apoc.graph.fromDocument($json, $config) yield graph",
                 Util.map("json", JsonUtil.OBJECT_MAPPER.writeValueAsString(artistGenesisMapExt), "config", Util.map("write", true)),
                 stringObjectMap -> {
                     Map<String, Object> map = stringObjectMap.next();
-                    assertEquals("Graph", ((Map) map.get("graph")).get("name"));
-                    Collection<Node> nodes = (Collection<Node>) ((Map) map.get("graph")).get("nodes");
-                    Collection<Relationship> relationships = (Collection<Relationship>) ((Map) map.get("graph")).get("relationships");
+                    assertEquals("Graph", ((Map<String, Object>) map.get("graph")).get("name"));
+                    Collection<Node> nodes = (Collection<Node>) ((Map<String, Object>) map.get("graph")).get("nodes");
+                    Collection<Relationship> relationships = (Collection<Relationship>) ((Map<String, Object>) map.get("graph")).get("relationships");
                     assertEquals(2, nodes.size());
                     assertEquals(1, relationships.size());
                     Iterator<Node> nodeIterator = nodes.iterator();
                     Iterator<Relationship> relationshipIterator = relationships.iterator();
 
                     Node albumGenesis = nodeIterator.next();
-                    assertEquals(asList(label("Album")), albumGenesis.getLabels());
-                    assertTrue(albumGenesis.getId() > 0);
+                    assertEquals(List.of(label("Album")), albumGenesis.getLabels());
+                    // Check this is not a virtual node
+                    assertTrue(!NumberUtils.isCreatable(albumGenesis.getElementId()) || Integer.parseInt(albumGenesis.getElementId()) > 0);
                     assertEquals(albumGenesisMap, albumGenesis.getAllProperties());
                     Node artistGenesis = nodeIterator.next();
-                    assertEquals(asList(label("Artist")), artistGenesis.getLabels());
-                    assertTrue(artistGenesis.getId() > 0);
+                    assertEquals(List.of(label("Artist")), artistGenesis.getLabels());
+                    // Check this is not a virtual node
+                    assertTrue(!NumberUtils.isCreatable(artistGenesis.getElementId()) || Integer.parseInt(artistGenesis.getElementId()) > 0);
```

Comment:
You could consider break out this to a small help method, something like

boolean nonVirtual(Entity entity) {
assertTrue(!NumberUtils.isCreatable(entity.getElementId()) || Integer.parseInt(entity.getElementId()) > 0);
}